api/candlepin-api-spec.yaml

```diff
@@ -1265,6 +1265,35 @@ components:
           type: string
           minLength: 1
 
+    EnvironmentDTO:
+      description: DTO representing an environment
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
+            id:
+              type: string
+            name:
+              type: string
+            description:
+              type: string
+            owner:
+              $ref: "#/components/schemas/NestedOwnerDTO"
+            environmentContent:
+              type: array
+              uniqueItems: true
```

Comment:
I hope this will not change the API schema defination. Previously it was not unique.

Reply:
It should be fine, because even thougn now it is a Collection<>, before the DTO was created we were using the Environment.java model entity, which uses a Set, so we should be fine

Reply:
No I don't think it will impact the APIs.
Earlier `environmentContent` property was backed by a HashMap (I assume this was done to gain performance with quick equality check while adding a content), now it is changed with a Set representation, this is much more aligned with the entity class `Environment` which also follows a Set representation for EnvironmentContent.

Reply:
While true, one thing we want to be careful about is adding restrictions that will inadvertently break current clients (even if they're doing silly things).

When it comes to API or communication design, I like to follow Postel's law as much as possible. Basically: be conservative in what you send, be liberal in what you accept.